api/operator/v1alpha1/status_test.go

```diff
@@ -285,3 +286,168 @@ func TestRemoveUnsupportedConditions(t *testing.T) {
 		})
 	}
 }
+
+func TestSetEventMeshAvailableConditionToTrue(t *testing.T) {
+	var (
+		anyCondition0 = kmetav1.Condition{
+			Type:    "any-type-0",
+			Status:  kmetav1.ConditionStatus("any-status-0"),
+			Reason:  "any-reason-0",
+			Message: "any-message-0",
+		}
+
+		anyCondition1 = kmetav1.Condition{
+			Type:    "any-type-1",
+			Status:  kmetav1.ConditionStatus("any-status-1"),
+			Reason:  "any-reason-1",
+			Message: "any-message-1",
+		}
+
+		anyCondition2 = kmetav1.Condition{
+			Type:    "any-type-2",
+			Status:  kmetav1.ConditionStatus("any-status-2"),
+			Reason:  "any-reason-2",
+			Message: "any-message-2",
+		}
+
+		backendAvailableConditionFalse = kmetav1.Condition{
+			Type:    string(ConditionBackendAvailable),
+			Status:  kmetav1.ConditionFalse,
+			Reason:  string(ConditionReasonBackendNotSpecified),
+			Message: ConditionBackendNotSpecifiedMessage,
+		}
+
+		backendAvailableConditionTrue = kmetav1.Condition{
+			Type:    string(ConditionBackendAvailable),
+			Status:  kmetav1.ConditionTrue,
+			Reason:  string(ConditionReasonEventMeshConfigAvailable),
+			Message: ConditionEventMeshConfigAvailableMessage,
+		}
+	)
+
+	tests := []struct {
+		name                string
+		givenEventingStatus EventingStatus
+		wantEventingStatus  EventingStatus
+	}{
+		// add new condition
+		{
+			name: "should add a new condition in case of nil conditions",
+			givenEventingStatus: EventingStatus{
+				Conditions: nil,
+			},
+			wantEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					backendAvailableConditionTrue,
+				},
+			},
+		},
+		{
+			name: "should add a new condition in case of empty conditions",
+			givenEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{},
+			},
+			wantEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					backendAvailableConditionTrue,
+				},
+			},
+		},
+		{
+			name: "should add a new condition and preserve existing ones",
+			givenEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					anyCondition0,
+					anyCondition1,
+					anyCondition2,
+				},
+			},
+			wantEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					anyCondition0,
+					anyCondition1,
+					anyCondition2,
+					backendAvailableConditionTrue,
+				},
+			},
+		},
+		// update existing condition
+		{
+			name: "should update existing condition",
+			givenEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					backendAvailableConditionFalse,
+				},
+			},
+			wantEventingStatus: EventingStatus{
+				Conditions: []kmetav1.Condition{
+					backendAvailableConditionTrue,
+				},
+			},
+		},
+		{
+			name: "should update condition and preserve existing ones",
```

Comment:
this test does not `update` anything

Reply:
Internally it will update the **LastTransitionTime** https://github.com/marcobebway/eventing-manager/blob/a954b017738de6e708d938d41a4e1f19d856cd91/api/operator/v1alpha1/status.go#L18.

Reply:
Well, then i have a wrong understanding. If the condition did not change (from true to true) imho it should NOT update the transition time. only if one of the values (reason, message etc) changed it transitioned from one state to another

Reply:
This will be fixed as part of https://github.com/kyma-project/eventing-manager/issues/542.